generate/generate.go

```diff
@@ -846,12 +850,20 @@ func (s *service) generateNewAPICallFunc(a *API) {
 	pn("		return nil, err")
 	pn("	}")
 	pn("")
-	if n == "CreateNetwork" {
+	if n == "CreateNetwork" || n == "RegisterTemplate" {
 		pn("	if resp, err = getRawValue(resp); err != nil {")
 		pn("		return nil, err")
 		pn("	}")
 		pn("")
 	}
+	if n == "RegisterTemplate" {
+		pn("	var l []json.RawMessage")
+		pn("	if err:= json.Unmarshal(resp, &l); err != nil {")
+		pn("		return nil, err")
+		pn("	}")
+		pn("	resp = l[0]")
+		pn("")
+	}	
```

Comment:
At this point I have a json list of templates in `resp`. (I am not sure why this comes back from Cloudstack, I expect always only a single template back in response)

Anyway, this fails when trying to unmarshal it directly into the RegisterTemplateResponse struct. so first, I just select the first template from the list here - this then populates the struct correctly